lib/ret/media_resolver.ex

```diff
@@ -103,15 +103,26 @@ defmodule Ret.MediaResolver do
     with ytdl_host when is_binary(ytdl_host) <- module_config(:ytdl_host) do
       case fetch_ytdl_response(query, ytdl_format) do
         %HTTPoison.Response{status_code: 500, body: body} ->
-          if String.contains?(body, "is offline") do
-            {:commit,
-             resolved(query.url, %{
-               expected_content_type: "text/html",
-               media_status: :offline_stream,
-               thumbnail: RetWeb.Endpoint.static_url() <> "/stream-offline.png"
-             })}
-          else
-            resolve_non_video(query, root_host)
+          # youtube-dl returns a 500 error, but includes the underlying error in its body text, search for some special cases
+          cond do
+            String.contains?(body, "is offline") ->
+              {:commit,
+               resolved(query.url, %{
+                 expected_content_type: "text/html",
+                 media_status: :offline_stream,
+                 thumbnail: RetWeb.Endpoint.static_url() <> "/stream-offline.png"
+               })}
+
+            String.contains?(body, "HTTPError 429") ->
```

Comment:
Probably unnecessary but I'd like to see an real example response from yt-dl when these errors happen (`"is offline"` or `"HTTPError 429"`). Sometime in the future their responses might change shape/form and we might more easily notice that kind of regression if we had full examples.

Reply:
Yep, as with most stuff with yotuube-dl, this is very fragile to upstream changes because its essentially just scraping. For this one, solong as the HTTP error code doesn't change I suspect this one is pretty safe, since this string we are searching for here comes from the Python http library. The "stream offline" one less so, since I think this is text coming from twitch. 

The full text I observed on a reticulum server that was being rate limited was:
> "error":"ERROR: Unable to download webpage: HTTP Error 429: Too Many Requests (caused by <HTTPError 429: 'Too Many Requests'>); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output."}

To test this I specifically set this string to the body text, since I don't have a good way of actually getting a server into this state.